components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+		}
+	}
+	return numDaysFromLastMonthToShowThisMonth;
+}
+
+function getNumberOfDaysInMonth(month, year) {
+	month++; // month index started at 0
+	let days = 31;
+	if (month === 2) {
+		if ((year % 4 === 0) && ((year % 100 !== 0) || (year % 400 === 0))) {
+			days = 29;
+		} else {
+			days = 28;
+		}
+	} else if (month === 4 || month === 6 || month === 9 || month === 11) {
+		days = 30;
+	}
+	return days;
+}
+
+function getNumberOfDaysToSameWeekPrevMonth(month, year, firstDayOfWeek) {
+	const prevMonth = getPrevMonth(month);
+	const prevMonthYear = prevMonth === 11 ? year - 1 : year;
+	const numDaysPrevMonth = getNumberOfDaysInMonth(prevMonth, prevMonthYear);
+	const numDaysFromPrevPrevMonthToShowPrevMonth = getNumberOfDaysFromPrevMonthToShow(prevMonth, prevMonthYear, firstDayOfWeek);
+	const numWeeksPrevMonth = Math.ceil((numDaysFromPrevPrevMonthToShowPrevMonth + numDaysPrevMonth) / 7);
+
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	const hasDaysFromPrevMonth = (firstDayOfMonth !== firstDayOfWeek);
+	return daysInWeek * (numWeeksPrevMonth - ((hasDaysFromPrevMonth) ? 1 : 0));
+}
+
+function getPrevMonth(month) {
+	return (month === 0) ? 11 : (month - 1);
+}
+
+class Calendar extends LocalizeStaticMixin(LitElement) {
+
+	static get properties() {
+		return {
+			selectedValue: { type: String, attribute: 'selected-value' },
+			summary: { type: String },
+			_shownMonth: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [calendarStyles, bodySmallStyles, heading4Styles];
+	}
+
+	static get resources() {
+		return {
+			'ar': { show: 'Show {month} ' },
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this._descriptor = getDateTimeDescriptor();
+
+		this._daysOfWeekIndex = [];
+		for (let i = this._descriptor.calendar.firstDayOfWeek; i < this._descriptor.calendar.firstDayOfWeek + daysInWeek; i++) {
+			this._daysOfWeekIndex.push(i % daysInWeek);
+		}
+
+		this._dialogLabelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.selectedValue = this.selectedValue ? new Date(this.selectedValue) : null;
+		this._today = new Date();
+		this._focusDate = this.selectedValue ? new Date(this.selectedValue.getFullYear(), this.selectedValue.getMonth(), this.selectedValue.getDate()) : new Date(this._today.getFullYear(), this._today.getMonth(), 1);
+		this._shownMonth = this.selectedValue ? this.selectedValue.getMonth() : this._today.getMonth();
+		this._shownYear = this.selectedValue ? this.selectedValue.getFullYear() : this._today.getFullYear();
+	}
+
```

Comment:
If consumers try to call `focus` on the component, I'm not sure it will do what we expect.  We _might_ need to implement `focus` and apply to the correct internal element.  There has been much discussion on focus behavior in web components, but it may still not be handled automatically.

Reply:
Good point - it does not seem to be handled automatically. My thought is that the element to receive focus in that case would be the left `d2l-button-icon`, since that's the way it is if you tab in. Does that make sense to you?

Reply:
Yep, I _think_ so.